app/services/work_packages/set_attributes_service.rb

```diff
@@ -43,6 +45,15 @@ def set_attributes(attributes)
     end
 
     set_custom_attributes(attributes)
+    mark_templated_subject
+  end
+
+  def mark_templated_subject
+    return true unless work_package.type&.replacement_patterns_defined?
+
+    if work_package.type.patterns.all_enabled[:subject]
+      work_package.subject = "Templated by #{work_package.type.name}"
```

Comment:
Yes. If there's no subject the contract will fail as it is a required field, so I fill in with something for now so that we can overwrite it later on.

It will get persisted as I might need "persisted" info (like `created_at`) for the pattern, but all is done within the context of the same request, so the user won't (hopefully) see it.

Reply:
Fair enough. Makes sense to me.

... I thought we might want to somehow tag these so that we can query any lingering/stuck cases later. But honestly `Templated by ` is probably good enough for that.